src/code/z_camera.c

```diff
@@ -2,13 +2,1791 @@
 #include <global.h>
 #include <vt.h>
 
-/*
-DATA
-*/
+typedef struct {
+    s16 val;
+    s16 param;
+} CameraModeValue;
+
+typedef struct {
+    s16 funcIdx;
+    s16 valueCnt;
+    CameraModeValue* values;
+} CameraMode;
+
+typedef struct {
+    union {
+        u32 unk_00;
+        struct {
+            u32 unk_bit0 : 1;
+            u32 unk_bit1 : 1;
+            u32 validModes : 30;
+        };
+    };
+    CameraMode* cameraModes;
+} CameraSetting;
+
+typedef union {
+    s16 sh[2];
+    s32 w;
+} u_D_8011D3A0;
+
+typedef enum {
+    CAM_SET_NONE,
+    CAM_SET_NORMAL0,
+    CAM_SET_NORMAL1,
+    CAM_SET_DUNGEON0,
+    CAM_SET_DUNGEON1,
+    CAM_SET_NORMAL3,
+    CAM_SET_HORSE0,
+    CAM_SET_BOSS_GOMA,
+    CAM_SET_BOSS_DODO,
+    CAM_SET_BOSS_BARI,
+    CAM_SET_BOSS_FGANON,
+    CAM_SET_BOSS_BAL,
+    CAM_SET_BOSS_SHADES,
+    CAM_SET_BOSS_MOFA,
+    CAM_SET_TWIN0,
+    CAM_SET_TWIN1,
+    CAM_SET_BOSS_GANON1,
+    CAM_SET_BOSS_GANON2,
+    CAM_SET_TOWER0,
+    CAM_SET_TOWER1,
+    CAM_SET_FIXED0,
+    CAM_SET_FIXED1,
+    CAM_SET_CIRCLE0,
+    CAM_SET_CIRCLE2,
+    CAM_SET_CIRCLE3,
+    CAM_SET_PREREND0,
+    CAM_SET_PREREND1,
+    CAM_SET_PREREND3,
+    CAM_SET_DOOR0,
+    CAM_SET_DOORC,
+    CAM_SET_RAIL3,
+    CAM_SET_START0,
+    CAM_SET_START1,
+    CAM_SET_FREE0,
+    CAM_SET_FREE2,
+    CAM_SET_CIRCLE4,
+    CAM_SET_CIRCLE5,
+    CAM_SET_DEMO0,
+    CAM_SET_DEMO1,
+    CAM_SET_MORI1,
+    CAM_SET_ITEM0,
+    CAM_SET_ITEM1,
+    CAM_SET_DEMO3,
+    CAM_SET_DEMO4,
+    CAM_SET_UFOBEAN,
+    CAM_SET_LIFTBEAN,
+    CAM_SET_SCENE0,
+    CAM_SET_SCENE1,
+    CAM_SET_HIDAN1,
+    CAM_SET_HIDAN2,
+    CAM_SET_MORI2,
+    CAM_SET_MORI3,
+    CAM_SET_TAKO,
+    CAM_SET_SPOT05A,
+    CAM_SET_SPOT05B,
+    CAM_SET_HIDAN3,
+    CAM_SET_ITEM2,
+    CAM_SET_CIRCLE6,
+    CAM_SET_NORMAL2,
+    CAM_SET_FISHING,
+    CAM_SET_DEMOC,
+    CAM_SET_UO_FIBER,
+    CAM_SET_DUNGEON2,
+    CAM_SET_TEPPEN,
+    CAM_SET_CIRCLE7,
+    CAM_SET_NORMAL4,
+    CAM_SET_MAX
+} CameraSettingType;
+
+typedef enum {
+    CAM_MODE_NORMAL,
+    CAM_MODE_PARALLEL,
+    CAM_MODE_KEEPON,
+    CAM_MODE_TALK,
+    CAM_MODE_BATTLE,
+    CAM_MODE_CLIMB,
+    CAM_MODE_SUBJECT,
+    CAM_MODE_BOWARROW,
+    CAM_MODE_BOWARROWZ,
+    CAM_MODE_FOOKSHOT,
+    CAM_MODE_BOOMERANG,
+    CAM_MODE_PACHINCO,
+    CAM_MODE_CLIMBZ,
+    CAM_MODE_JUMP,
+    CAM_MODE_HANG,
+    CAM_MODE_HANGZ,
+    CAM_MODE_FREEFALL,
+    CAM_MODE_CHARGE,
+    CAM_MODE_STILL,
+    CAM_MODE_PUSHPULL,
+    CAM_MODE_BOOKEEPON
+} CameraModeType;
+
+/*==================================================================*/
+// Data
+s16 sOREGInit[] = {
+    0x0000, 0x0001, 0x0005, 0x0005, 0x0005, 0x38A4, 0x0014, 0x0010, 0x0096, 0x0019, 0x0096, 0x0006, 0x000A, 0x000A,
```

Comment:
Decimal should probably be used here, and there are also a bunch of negative values in here that should be fixed.